### PR TITLE
Added thicken and new split tools

### PR DESCRIPTION
**Pull request by Jojain**
I redid the PR on another branch, was simpler than to solve the other one

### CODE REVIEW

File: src/build123d/topology.py [hunk 1]
```diff
@@ -366,6 +366,7 @@
 
 Shapes = Literal["Vertex", "Edge", "Wire", "Face", "Shell", "Solid", "Compound"]
 
+TrimmingTool = Union[Plane,"Shell", "Face"]
```
Comment (jdegenstein) [inline src/build123d/topology.py]: I think there is a space missing between `Plane, "Shell"` (just don't want this to randomly show up in a future unrelated PR).

**Comment by jdegenstein** [comment]
Looks like there are a few formatting inconsistencies, I left one comment but found more in the added tests. Can you run try running black again on that file too? Hope it doesn't cause more issues. Also one test should be "planar" instead of plarnar.

File: src/build123d/topology.py [hunk 2]
```diff
@@ -7000,6 +6982,26 @@ def make_loft(
...
         return cls(_make_loft(objs, False, ruled))
 
+    def thicken(self, depth: float) -> Solid:
+        """Thicken Shell
+
+        Create a solid from a shell by thickening along the normals.
+
+        Args:
+            depth (float): Amount to thicken face(s), can be positive or negative.
+            normal_override (Vector, optional): The normal_override vector can be used to
```
Comment (gumyr) [inline src/build123d/topology.py]: The `Shell` version of `thicken` doesn't have the `normal_override` parameter so it needs to be removed from the docstring.

**Comment by gumyr** [comment]
Thanks!

**Pull request merged**